### ISSUE TITLE
Receiving a HTTP 409 Conflict response when trying to rename a file with OneDrive Personal

### ISSUE THREAD

**Issue by kotsak** [OneDrive/onedrive-api-docs#767]
Hi,

We have a OneDrive Personal user who receives a *HTTP/1.1 409 Conflict* error when our application is trying to rename any file (and set its last modified timestamp) it has just finished uploading to the server.

Below are the request and response packets:

```
PATCH /v1.0/drive/items/B0E7B9973C67B12B!466303 HTTP/1.1
Content-Type: application/json; charset=UTF-8
Content-Length: 105
Authorization: Bearer xxxxxxxxx
Host: api.onedrive.com
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
User-Agent: Mozilla/3.0 (compatible; Indy Library)

{"name":"Annual Privacy Notice.doc","fileSystemInfo":{"lastModifiedDateTime":"2011-02-22T12:34:08.560Z"}}

HTTP/1.1 409 Conflict
Content-Length: 90
Content-Type: application/json
P3P: CP="BUS CUR CONo FIN IVDo ONL OUR PHY SAMo TELo"
X-MSNSERVER: BN2BAPF0E7098F2
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-QosStats: {"ApiId":0,"ResultType":2,"SourcePropertyId":0,"TargetPropertyId":42}
X-ThrowSite: 0bfa.2610
X-AsmVersion: UNKNOWN; 19.88.1204.2027
X-MSEdge-Ref: Ref A: 06167BAFEC674796B238B7670371BEA9 Ref B: ATAEDGE0622 Ref C: 2018-01-06T18:09:57Z
Date: Sat, 06 Jan 2018 18:09:57 GMT

{"error":{"code":"resourceModified","message":"ETag does not match current item's value"}}
```
As you can see It's clear that we don't provide any ETag along with the request. We have tried to reproduce this on our side using different accounts but it always works as expected.

We are suspecting the problem is particular to this user's account so maybe you can tell us why it fails.

Thanks,

**Comment by ificator** [comment]
Do you know if the user has lots of other actions being performed on their OneDrive at the time?

At the very least the error response is misleading, as you quite obviously did not include an ETag.

**Comment by kotsak** [comment]
Thank you for looking into this. We've asked the user and he said that there was no other activity involving his OneDrive account at that time.

**Comment by ificator** [comment]
Thanks @kotsak - is it possible that your application is executing multiple uploads in parallel? Also, the pattern I see appears to be the following - can you confirm?
1. POST to /children
2. PATCH to item created in 1
3. DELETE when PATCH fails

**Comment by kotsak** [comment]
It is possible for the user to create different profiles in our application targetting the cloud storage and set them to run in parallel but each profile is run in a single thread so it's not possible to execute multiple uploads within the same profile and all API calls are in sequence. When asked, the user said that no other activity or profile was running at the time except the one that gave the error.

As for the pattern, you are correct, except in cases where a large file may get uploaded so in this case step(1) will involve calls related to large file upload (create session, upload, etc) and then steps (2) & (3) will follow.

**Comment by ificator** [comment]
My only theory is that the rename is racing with an operation that is taking place behind the scenes on newly uploaded photos. We should be able to tell if that's the case by comparing the `eTag` returned from step (1) with the `eTag` returned from a `GET` request after step (2) failed. I'm not sure how easy it would be for you to collect that data though since that pattern isn't what your app does.

I guess you could add the `GET` to confirm the state after failure and then another `PATCH` to rename if things look as expected. That retry might also solve your problem, but it would obviously be nice if it wasn't required.

**Comment by song09585** [comment]
We have conflict issue too when trying to upload a xlsx file sometimes.
We sending a upload request with "name.conflictBehavior=replace" option, but server reports 
**http code : 409
error message: ETag does not match current item's value**

does the "name.conflictBehavior=replace" option not work?

**Comment by AnnieChmarak** [comment]
One of our clients faces the same issue. Our application does the following:

1. uploads a file to user's OneDrive (PUT)
2. sets fileSystemInfo to this file's metadata (PATCH)

None of theese uses eTag.

The user processed lots of lightweighted images and sometimes PATCH requests ended up with the error described.
The application executes requests one-by-one.

As I can see, a workaround is just to retry patching.
Can I be helpful in finding the cause?

**Comment by dstshared** [comment]
Is there any update about this issue?
We still hitting this issue when trying to upload  xlsx files.

the server response:
{"error":{"code":"resourceModified","message":"The resource being updated has changed since the caller last read it, usually an eTag mismatch."}

**Comment by patrick-rodgers** [comment]
As part of a repository clean up effort we are closing older issues. If this issue remains, please: open a new issue, reference this issue, and provide any additional details that may help in resolution. Thank you for your understanding as we work to improve our responsiveness.
